integration-tests/src/test/java/org/apache/fineract/integrationtests/AdvancedPaymentAllocationLoanRepaymentScheduleTest.java

```diff
@@ -4113,6 +4114,67 @@ public void uc137() {
+    // ADVANCED_PAYMENT_ALLOCATION_STRATEGY
+    // 1. Create a Loan product, Loan Progressive type and other transaction processing strategy code
+    // different than advanced-payment-allocation-strategy
+    // - > Expect validation error
+    @Test
+    public void uc142() {
+        runAt("15 January 2023", () -> {
+            Integer numberOfRepayments = 4;
+            double amount = 1000.0;
+            String loanDisbursementDate = "1 January 2023";
+
+            Long clientId = clientHelper.createClient(ClientHelper.defaultClientCreationRequest()).getClientId();
+            final BigDecimal interestRatePerPeriod = BigDecimal.valueOf(2.5);
+
+            LOG.info("------------------------------CREATING NEW LOAN PRODUCT ---------------------------------------");
+            PostLoanProductsResponse loanProductResponse = loanProductHelper
+                    .createLoanProduct(createOnePeriod30DaysLongNoInterestPeriodicAccrualProductWithAdvancedPaymentAllocation()
+                            .addCreditAllocationItem(createChargebackAllocation())
+                            .loanScheduleType(LoanScheduleType.PROGRESSIVE.toString()));
+
+            Long loanId = applyAndApproveLoanProgressiveAdvancedPaymentAllocationStrategyMonthlyRepayments(clientId,
+                    loanProductResponse.getResourceId(), numberOfRepayments, loanDisbursementDate, amount, interestRatePerPeriod);
+
+            verifyRepaymentSchedule(loanId, //
+                    installment(1000.0, null, "01 January 2023"), //
+                    installment(248.88, false, "01 February 2023"), //
+                    installment(249.56, false, "01 March 2023"), //
+                    installment(249.94, false, "01 April 2023"), //
+                    installment(251.62, false, "01 May 2023") //
+            );
+
+            GetLoansLoanIdResponse loanDetails = loanTransactionHelper.getLoanDetails(loanId);
+            assertNull(loanDetails.getSummary());
+
+            loanTransactionHelper.disburseLoan(loanId, new PostLoansLoanIdRequest().actualDisbursementDate("1 January 2023")
+                    .dateFormat(DATETIME_PATTERN).transactionAmount(BigDecimal.valueOf(1000.0)).locale("en"));
+
+            Long repayment1TransactionId = addRepaymentForLoan(loanId, 251.0, "2 January 2023");
+            assertNotNull(repayment1TransactionId);
+            loanDetails = loanTransactionHelper.getLoanDetails(loanId);
+            LOG.info("Data {}", loanDetails.getSummary().getTotalUnpaidAccruedDueInterest());
+            LOG.info("Data {}", loanDetails.getSummary().getTotalUnpaidAccruedNotDueInterest());
+
+            Long repayment2TransactionId = addRepaymentForLoan(loanId, 251.0, "3 January 2023");
+            assertNotNull(repayment2TransactionId);
+            loanDetails = loanTransactionHelper.getLoanDetails(loanId);
+            LOG.info("Data {}", loanDetails.getSummary().getTotalUnpaidAccruedDueInterest());
```

Comment:
Writing out with logger is not enough! Please user assertions to validate the proper amount was calculated and retrieved!